### ISSUE TITLE
Some ideas for improvements

### ISSUE THREAD

**Issue by kryostase** [Forien/foundryvtt-forien-unidentified-items#2]
Thx for the module, Forien. It's great!

Some ideas that came into my mind, when I was playing around with it.
1. Would be great if you can configure options in the module setting menu like: 
- Keep original image of the item
- Set item type or even category as default name. So a Lightning Sword of Xorhash will be just a Sword
- a) Duplicate the item (like it is right now) OR  b) Don't duplicate it (delete the original item) if thats possible. Or just create a new unidentified item and when it is identified replace it by the linked original item.

2. As many of us using loot sheet characters. Would be great if you can do the mystify right from a character not only from the sidebar item menu

**Comment by Forien** [comment]
@kryostase thanks for your kind words!

As for suggestions:   
> Keep original image of the item  

This would probably be needed to be on per-item basis. Sometimes image spoils what it is. You can still edit the mystified item to set any image you want. And there will be option to "Mystify as..." where you can mystify one item, but take appearance and base data from other item (so `Magical Sword of Burning +10` could be mystified as `Plain Longsword` and work exactly as `Plain Longsword` until identified). 

> Set item type or even category as default name. So a Lightning Sword of Xorhash will be just a Sword

Such things are system-specific, module aims to be system agnostic. Some system integration will happen, but for example dnd5e and wfrp4e do not have distinct "sword" category. So it will require either me to make custom integration for given system. 

> Don't duplicate it (delete the original item) if thats possible. 

That would be possible
> when it is identified replace it by the linked original item.

The reason I went with copying data during mystification is that so you can delete/edit the original item without causing issues with unidentified one. By linking item, deleting original should also delete mystified one. That adds too much unneeded complexity and places for module to break ;) 

> As many of us using loot sheet characters. Would be great if you can do the mystify right from a character not only from the sidebar item menu

Well, you can if you open the item, but it will then create new one in sidebar. I don't know exactly how Loot Sheet works, but I assumes it's publicly visible to players. 

Regardless, making option to "mystify as replacement" (so mystifying the original without creating copy), will automatically allow to mystify Actor's OwnedItems (including Loot Sheet). 


### So, to sum this up:
- I will add option to "Mystify as replacement", so instead of creating Unidentified copy, original item will be altered. 
- As stated in README I will try to bring some basic system integration, at least for dnd5e and wfrp4e

Thank you for providing feedback and suggestions and thank you for trying out this module! Please consider dropping by on my [Patreon page](https://www.patreon.com/forien) and supporting module development if you can.   
Also, on my Patreon page, there will always be public posts about new releases, major patch notes etc.

**Comment by kryostase** [comment]
Thx for your response! And thx for some of the changes.

One thing you mentioned is at least not working in PF2e. I posted a new bug issue for that one.

**Comment by Forien** [comment]
I was the new bug, it was duplicate of other issue posted here. I renamed the original issue so it should be easier to spot. Right now, mystifying works only from sidebar. I will fix it, I just need to finish the Quest Log v0.4.0 first :/

**Comment by MSAbaddon** [comment]
Concerning the idea of keeping the original image - you could just let the GM choose if he wants to keep the image or not when he klicks on mystify. The idea of "mystify as" is nice for many items such as weapons and armor, but wondrous items are very specific and instead of mysifying them into a different item, it would - in my opinion, be great to keep the original image.

**Comment by Forien** [comment]
> Concerning the idea of keeping the original image - you could just let the GM choose if he wants to keep the image or not when he klicks on mystify. The idea of "mystify as" is nice for many items such as weapons and armor, but wondrous items are very specific and instead of mysifying them into a different item, it would - in my opinion, be great to keep the original image.

I am working on doing 3 options:
1. `Mystify As...` – I think explained already
1. `Replace with mystified` – will mystify current item, not create copy
1. `Mystify...` – currently I have no better idea for name of it, but it will popup window full of checkboxes to allow GM choose which properties (including name/image etc) to keep.

So I hope that will adress your request ;)

**Comment by MSAbaddon** [comment]
Ok, that's great :-)
Thanks for the awesome work you do
